packages/design-system/src/components/Accordion/Accordion.tsx

```diff
@@ -1,148 +1,357 @@
+  }
+
+  &:last-child {
+    border-bottom-left-radius: ${(props) => (props.$size === 'S' ? '0.3rem' : '')};
+    border-bottom-right-radius: ${(props) => (props.$size === 'S' ? '0.3rem' : '')};
+  }
+
+  & + & {
+    border-top: ${(props) => (props.$size === 'S' ? `solid 1px ${props.theme.colors.neutral200}` : 'unset')};
+  }
+
+  /* This applies our desired focus effect correctly. */
+  &:focus-within {
+    position: relative;
+    z-index: 1;
+    box-shadow: 0 0 0 2px ${(props) => props.theme.colors.primary600};
+  }
+`;
+
+/* -------------------------------------------------------------------------------------------------
+ * Trigger
+ * -----------------------------------------------------------------------------------------------*/
+
+type TriggerElement = HTMLButtonElement;
+
+interface TriggerProps extends Omit<RadixAccordion.AccordionTriggerProps, 'asChild'> {
   /**
-   * Color variant for `Accordion`. The "secondary" variant reverses the background colors.
-   * This is useful when you want to display a list and alternate the colors of the accordions.
+   * @default "left"
    */
-  variant?: AccordionVariant;
+  caretPosition?: 'left' | 'right';
+  description?: string;
+  icon?: React.ElementType;
 }
 
-const Accordion = ({
-  children,
-  disabled = false,
-  error,
-  expanded = false,
-  hasErrorMessage = true,
-  id,
-  onToggle,
-  size = 'M',
-  variant = 'primary',
-  shadow,
-}: AccordionProps) => {
-  const generatedId = useId(id);
-
-  const context = React.useMemo(
-    () => ({ expanded, onToggle, id: generatedId, size, variant, disabled }),
-    [disabled, expanded, generatedId, onToggle, size, variant],
-  );
-
-  return (
-    <AccordionContext.Provider value={context}>
-      <AccordionWrapper
-        data-strapi-expanded={expanded}
-        aria-disabled={disabled}
-        hasRadius
-        shadow={shadow}
-        $expanded={expanded}
-        $disabled={disabled}
-        $variant={variant}
-        $error={error}
-      >
-        {children}
-      </AccordionWrapper>
-      {error && hasErrorMessage && (
-        <Box paddingTop={1}>
-          <Typography variant="pi" textColor="danger600">
-            {error}
-          </Typography>
-        </Box>
-      )}
-    </AccordionContext.Provider>
-  );
-};
+const Trigger = React.forwardRef<TriggerElement, TriggerProps>(
+  ({ caretPosition = 'left', description, icon: Icon, children, ...restProps }, forwardedRef) => {
+    const { size } = useAccordion('Trigger');
 
-type AccordionWrapperProps = Pick<AccordionProps, 'expanded' | 'disabled' | 'variant' | 'error'>;
+    return (
+      <AccordionTrigger $caretPosition={caretPosition} $size={size} ref={forwardedRef} {...restProps}>
+        {caretPosition === 'left' ? (
+          <TriggerIcon $size={size}>
+            <CaretDown width={size === 'S' ? '1.2rem' : '1.6rem'} height={size === 'S' ? '1.2rem' : '1.6rem'} />
+          </TriggerIcon>
```

Comment:
Is it worth having this as a const rather than repeating it for left and right?

Reply:
we could, it's a much of a muchness... I don't think we'll be messing with this at all and it's pretty simple